tests/test_diaSourceMetrics.py

```diff
@@ -0,0 +1,50 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+import lsst.utils.tests
+from astropy.table import Table
+from lsst.analysis.tools.atools import diaSourceMetrics as diaAtool
+from lsst.analysis.tools.contexts import DrpContext
+
+
+class DiaSourceTest(lsst.utils.tests.TestCase):
+    """Test to see if DiaSources are counted as expected
+    and a plot is generated without falling over."""
+
+    def setUp(self):
+        testFile = "tests/diaSourceTable_tract_test.ecsv"
+        self.data = Table.read(testFile)
```

Comment:
Please make a new test file that's called `assocDiaSrc_test.ecsv` or similar. I made this original one by loading up an instance of the data product in question with the Butler (in this case it's a `goodSeeingDiff_assocDiaSrc`) and writing out 100 lines or so to an ecsv file via astropy Table export. I just want to make sure we aren't accidentally testing against a different table schema, since the one you're using here is a DRP data product I use for another test.

Reply:
Done. I've left a comment on the ticket explaining what dataset I used, and from which repo and collection (for future reference and reproducability purposes).